File: ios/ExpoNosyLoggerModule.swift
import ExpoModulesCore
import NosyLogger

public class ExpoNosyLoggerModule: Module {

    private let logger = NosyLogger()

    public func definition() -> ModuleDefinition {
        Name("ExpoNosyLogger")

        Function("init") { (apiKey: String) -> Void in
            logger.start(apiKey: apiKey)
        }

        Function("debug") { (message: String) -> Void in
            logger.debug(message)
        }

        Function("info") { (message: String) -> Void in
            logger.info(message)
        }

        Function("warning") { (message: String) -> Void in
            logger.warning(message)
        }

        Function("error") { (message: String) -> Void in
            logger.error(message)
        }
    }
}

File: android/src/main/java/dev/nosytools/rnlogger/expo/ExpoNosyLoggerModule.kt
package dev.nosytools.rnlogger.expo

import android.content.pm.PackageManager
import dev.nosytools.logger.Logger
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoNosyLoggerModule : Module() {

    private val context
        get() = requireNotNull(appContext.reactContext)

    private val apiKey by lazy {
        val applicationInfo =
                context.packageManager.getApplicationInfo(
                        context.packageName.toString(),
                        PackageManager.GET_META_DATA
                )

        applicationInfo.metaData.getString("NOSY_LOGGER_API_KEY")
                ?: throw IllegalStateException(
                        "Api Key was not found - set it via plugin configuration"
                )
    }

    private val logger by lazy { Logger(context).apply { init(apiKey) } }

    override fun definition() = ModuleDefinition {
        Name("ExpoNosyLogger")

        Function("debug") { message: String -> logger.debug(message) }

        Function("info") { message: String -> logger.info(message) }

        Function("warning") { message: String -> logger.warning(message) }

        Function("error") { message: String -> logger.error(message) }
    }
}
